src/alda/AldaServer.java

```diff
@@ -270,12 +287,18 @@ public void play(String code, String history, String from, String to)
     AldaResponse res = req.send(3000, 0);
 
     if (!res.success) {
-      error(res.body);
+      if (catchExceptions)
+        error(res.body);
+      else
+        throw new ServerRuntimeError(res.body);
       return;
     }
 
     if (res.workerAddress == null) {
-      error("No worker address included in response; unable to check for status.");
+      if (catchExceptions)
+        error("No worker address included in response; unable to check for status.");
+      else
+        throw new ServerRuntimeError("No worker address included in response; unable to check for status.");
```

Comment:
There is some repeated code here -- I wonder if it would make sense for the `error` function to check `catchExceptions` and either print the error message or throw a ServerRuntimeError?